ports/nrf/common-hal/bleio/CharacteristicBuffer.c

```diff
@@ -0,0 +1,75 @@
+/*
+ * This file is part of the MicroPython project, http://micropython.org/
+ *
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2018 Artur Pacholec
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+#include <string.h>
+#include <stdio.h>
+
+#include "ble_drv.h"
+#include "ble_gatts.h"
+#include "nrf_soc.h"
+
+#include "py/runtime.h"
+
+#include "common-hal/bleio/__init__.h"
+#include "common-hal/bleio/CharacteristicBuffer.h"
+
+STATIC void characteristic_buffer_on_ble_evt(ble_evt_t *ble_evt, void *param) {
+    bleio_characteristic_buffer_obj_t *self = (bleio_characteristic_buffer_obj_t *) param;
+    switch (ble_evt->header.evt_id) {
+    case BLE_GATTS_EVT_WRITE: {
```

Comment:
A switch with one case seems weird. Either it should be an if or have cases that do nothing for now.

Reply:
The `xxx_on_ble_evt` handlers are all this style. A lot of events are ignored. As we handle more events, we'll add more cases. I expect there to be more cases here eventually.

The event handlers are chained. An event is delivered to all of the event handlers in succession.